prov/rxd/src/rxd_av.c

```diff
@@ -185,25 +142,48 @@ static int rxd_av_insert(struct fid_av *av_fid, const void *addr, size_t count,
 	}
 
 	for (; i < count; i++, addr = (uint8_t *) addr + av->dg_addrlen) {
-		node = ofi_rbmap_find(&av->rbmap, (void *) addr);
-		if (node) {
-			rxd_addr = (fi_addr_t) node->data;
-		} else {
-			ret = rxd_av_insert_dg_addr(av, addr, &rxd_addr,
-						    flags, context);
-			if (ret)
-				break;
-		}
-
-		util_addr = av->rxd_addr_table[rxd_addr].fi_addr == FI_ADDR_UNSPEC ?
-			    rxd_set_fi_addr(av, rxd_addr) :
-			    av->rxd_addr_table[rxd_addr].fi_addr;
+		
+        ret = rxd_av_insert_dg_addr(av, addr, &dg_addr, flags, context);
+        if (ret)
+            break;
+        ret = rxd_av_insert_fi_addr(av, &dg_addr, &util_addr , flags);
+        if (ret) {
+            fi_av_remove(av->dg_av, &dg_addr, 1, flags);
```

Comment:
See earlier comment about just calling ofi_av_insert_addr here since you already handle the remove case

Reply:
Also probably need to handle the return here appropriately